doc/sphinx/proof-engine/tactics.rst

```diff
@@ -1410,94 +1426,101 @@ analysis on inductive or co-inductive objects (see :ref:`inductive-definitions`)
-   name.
+      This behaves as :n:`destruct @term` but adds an equation
+      between :token:`term` and the value that it takes in each of the
+      possible cases. The name of the equation is specified
+      by :token:`naming_intro_pattern` (see :tacn:`intros`),
+      in particular ``?`` can be used to let Coq generate a fresh name.
 
-.. tacv:: destruct @term with @bindings_list
+   .. tacv:: destruct @term with @bindings_list
 
-   This behaves like :n:`destruct @term` providing explicit instances for the
-   dependent premises of the type of :n:`@term` (see :ref:`syntax of bindings <bindingslist>`).
+      This behaves like :n:`destruct @term` providing explicit instances for
+      the dependent premises of the type of :token:`term`.
 
-.. tacv:: edestruct @term
-   :name: edestruct
+   .. tacv:: edestruct @term
+      :name: edestruct
 
-   This tactic behaves like :n:`destruct @term` except that it does not fail if
-   the instance of a dependent premises of the type of :n:`@term` is not
-   inferable. Instead, the unresolved instances are left as existential
-   variables to be inferred later, in the same way as :tacn:`eapply` does.
+      This tactic behaves like :n:`destruct @term` except that it does not
+      fail if the instance of a dependent premises of the type
+      of :token:`term` is not inferable. Instead, the unresolved instances
+      are left as existential variables to be inferred later, in the same way
+      as :tacn:`eapply` does.
 
-.. tacv:: destruct @term using @term
-.. tacv:: destruct @term using @term with @bindings_list
+   .. tacv:: destruct @term using @term {? with @bindings_list }
 
-   These are synonyms of :n:`induction @term using @term` and
-   :n:`induction @term using @term with @bindings_list`.
+      This is synonym of :n:`induction @term using @term {? with @bindings_list }`.
 
-.. tacv:: destruct @term in @goal_occurrences
+   .. tacv:: destruct @term in @goal_occurrences
 
-   This syntax is used for selecting which occurrences of :n:`@term` the case
-   analysis has to be done on. The :n:`in @goal_occurrences` clause is an
-   occurrence clause whose syntax and behavior is described in
-   :ref:`occurences sets <occurencessets>`.
+      This syntax is used for selecting which occurrences of :token:`term`
+      the case analysis has to be done on. The :n:`in @goal_occurrences`
+      clause is an occurrence clause whose syntax and behavior is described
+      in :ref:`occurences sets <occurencessets>`.
 
-.. tacv:: destruct @term with @bindings_list as @disj_conj_intro_pattern eqn:@naming_intro_pattern using @term with @bindings_list in @goal_occurrences
-.. tacv:: edestruct @term with @bindings_list as @disj_conj_intro_pattern eqn:@naming_intro_pattern using @term with @bindings_list in @goal_occurrences
+   .. tacv:: destruct @term {? with @bindings_list } {? as @disj_conj_intro_pattern } {? eqn:@naming_intro_pattern } {? using @term {? with @bindings_list } } {? in @goal_occurrences }
```

Comment:
Doesn't this now subsume the previous variant?

Reply:
I'm unsure what you mean by this. The point of this variant is to show the general form where all the possible previous variants are available, so yes in a sense it subsumes them, but the documentation of these variants is to be found above.